### ISSUE TITLE
Running more tests with admission webhook (validation) enabled

### ISSUE THREAD

**Issue by programmer04** [Kong/kubernetes-ingress-controller#4680]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Problem Statement

Currently, the admission webhook is tested in the integration suite in isolation in files:
- `gateway_webhook_test.go`
- `httproute_webhook_test.go`
- etc.

But tests that apply actual configurations (e.g. routing traffic based on Ingress configuration, etc.) and test their correctness don't have webhook enabled. Every valid configuration should not be blocked by webhook too.  On the other hand, due to the nature of K8s, the admission webhook may be not configured/removed thus KIC should cope with invalid configurations too (webhook should be able to reject as many as possible). It has to be tested.

### Proposed Solution

Treat admission webhook enabled as the default configuration for KIC, and run as much as possible test with it enabled. 

### Additional information

It's been discovered during the work on 
- https://github.com/Kong/kubernetes-ingress-controller/pull/4608

that implemented (and covered with tests) features used to be rejected by [admission webhook as not implemented yet](https://github.com/Kong/kubernetes-ingress-controller/pull/4608/files#diff-0c50923b5a52b9ed325472451a7dad9051ac4e726d1c2afda92c284a295583b0L88-L101). Another complementary issue has been created too
- https://github.com/Kong/kubernetes-ingress-controller/issues/4676

### Acceptance Criteria

- [ ] All tests that expect to have valid configuration should go through webhook
- [ ] Invalid configs should be tested both with webhook enabled (config is rejected and not applied) and disabled (config is applied, KIC handles it)

**Comment by pmalek** [comment]
I believe we could actually investigate the possibility of running webhook tests with envtest (and perhaps Kong Gateway started via `testcontainers`).

With the introduction of Gateway Discovery tests like admission webhook integration tests might become more and more complicated to setup and debug.

**Comment by czeslavo** [comment]
> I believe we could actually investigate the possibility of running webhook tests with envtest (and perhaps Kong Gateway started via `testcontainers`).
> 
> With the introduction of Gateway Discovery tests like admission webhook integration tests might become more and more complicated to setup and debug.

+1 
I had an attempt to adapt our admission server boilerplate to `controller-runtime` so it's run as `Manager`'s runnable, not separately as it is now. It would allow running it in envtests more easily. Leaving the WIP PR for posterity: https://github.com/Kong/kubernetes-ingress-controller/pull/4763

**Comment by czeslavo** [comment]
> What would be the difference in tests? Convenience how it's started?

Yes, I believe it should allow using `envtest` [webhook install options](https://github.com/kubernetes-sigs/controller-runtime/blob/c30c66d67f47feeaf2cf0816e11c6ec0260c6e55/pkg/envtest/webhook.go#L42) with less friction.

**Comment by pangruoran** [comment]
Currently, we use the `ensureAdmissionRegistration`  function to registe the Admission webhook.

https://github.com/Kong/kubernetes-ingress-controller/blob/9a38f984baed428847e0e7f277ff6eb2a5bcfbb7/test/integration/webhook_test.go#L603

And call it, when we want to test with the Admission webhook.

* https://github.com/Kong/kubernetes-ingress-controller/blob/9a38f984baed428847e0e7f277ff6eb2a5bcfbb7/test/integration/gateway_webhook_test.go#L32-L47
* https://github.com/Kong/kubernetes-ingress-controller/blob/9a38f984baed428847e0e7f277ff6eb2a5bcfbb7/test/integration/httproute_webhook_test.go#L216-L231

I think we can have several options:

*  Add it as an addon to KTF.
    * I think there will be relatively few points of modification.

* Explore the method mentioned by @czeslavo  for [webhook install options](https://github.com/kubernetes-sigs/controller-runtime/blob/c30c66d67f47feeaf2cf0816e11c6ec0260c6e55/pkg/envtest/webhook.go#L42)

**Comment by pangruoran** [comment]
Another option:

We can deploy the resources we need through Helm, which can also simplify many steps.

* https://github.com/Kong/kubernetes-ingress-controller/issues/4973

I will temporarily leave this issue, I think #4973 will be a good choice.
